plugin.rb

```diff
@@ -108,4 +108,16 @@ module ::Follow
     next if !new_record
     Follow::NotificationHandler.new(post, notified).handle
   end
+
+  filter_following_topics = ->(scope, username, guardian) do
+    user = User.find_by(username: username)
+
+    next scope if user.nil?
+    next scope.none if user.id != guardian.user.id && !guardian.user.staff?
+
+    topic_ids = UserFollower.posts_for(user, current_user: guardian.user).map { |p| p.topic_id }
```

Comment:
Any chances we could write this as an SQL query instead of loading all the posts model in Ruby-land just to retrieve the topic ids?

Reply:
I've added a `topics_for` **_but_** I had thought of just adding:

```rb
    topic_ids =
      UserFollower
        .posts_for(user, current_user: guardian.user)
        .select("DISTINCT topic_id")
        .pluck(:topic_id)
```

But posts_for preloads posts so...

But I can change if you want